src/Storages/MergeTree/IMergeTreeDataPart.cpp

```diff
@@ -1341,11 +1341,11 @@ void IMergeTreeDataPart::loadColumns(bool require)
         .choose_kind = false,
     };
 
-    SerializationInfoByName infos(loaded_columns, settings);
+    SerializationInfoByName infos;
     if (metadata_manager->exists(SERIALIZATION_FILE_NAME))
```

Comment:
is it intended that now if in line 1345 condition is not satisfied then infos is not initialised?

Reply:
Yes, it's ok. If `SerializationInfoByName` is not initialized the default serialization will be returned for all columns. It's what expected.